priv/cabbage/apps/itest/test/features/in_flight_exits.feature

```diff
@@ -14,3 +14,24 @@ Feature: In Flight Exits
     And Bob starts a piggybacked in flight exit using his most recently prepared in flight exit data
     And Alice fully challenges Bobs most recent invalid in flight exit
     Then Alice can processes her own most recent in flight exit
+
+  Scenario: In-flight exited transaction input is not spendable
+    Given "Alice" deposits "10" ETH to the root chain
+    Then "Alice" should have "10" ETH on the child chain after finality margin
+    Given Alice creates a transaction for "5" ETH
+    And Alice starts an in flight exit from the most recently created transaction
+    Then Alice verifies its in flight exit from the most recently created transaction
+    Given Alice piggybacks output from her most recent in flight exit
+    And "Alice" in flight transaction inputs are not spendable any more
+    Then Alice can processes her own most recent in flight exit
+
+  Scenario: In-flight exited transaction piggybacked output is not spendable
+    Given "Alice" deposits "10" ETH to the root chain
+    Then "Alice" should have "10" ETH on the child chain after finality margin
+    Given Alice creates a transaction for "5" ETH
+    And Alice sends the most recently created transaction
+    And Alice starts an in flight exit from the most recently created transaction
+    Then Alice verifies its in flight exit from the most recently created transaction
+    Given Alice piggybacks output from her most recent in flight exit
+    And "Alice" in flight transaction most recently piggybacked output is not spendable any more
+    Then Alice can processes her own most recent in flight exit
```

Comment:
It would be nice to have a test where:
1. ife and piggyback come first
2. then comes the ch-ch block with a transaction that created the piggybacked output
3. piggybacked output is not available to spend
it might be hard to implement that in cabbage, but maybe a test involving just the db would do?

Reply:
Yes - that's exactly the egde case I included handling the on-finalization event.
We don't have such control of the events in cabbage test - I'll include it elsewhere and point here. Thanks :+1:

Reply:
I've added ExitConsumer tests, however testing on DB level above scenario with all 3 steps above will be unnecessary bloat because in pt 1. we know we are trying to spent non-existing output then in pt 2. we're adding it and finally spending in 3. 

I've check these points in the separate tests.

Reply:
I think test we have in `OMG.WatcherInfo.ExitConsumerTest` do not add up to the scenario I posted.
We do not have a test where event comes before the utxo.
If this is implied from the tests we have, please put a comment explaining how that works.

Reply:
Reasoning from what I can find in the code it looks like it will work well but tests would still be good.
And I still do not get why we need spending on finalization. I think all would work well without it. Could you please explain why we need that?

Reply:
In the scenario you posted:
1. Watcher-Info receives an event IFE-started and speds inputs
2. Watcher-Info receives an event Output piggybacked, but block including in-flight tx has not been received yet (👇 next point)
3. Watcher-Info receives an event New block - which creates an in-flight tx outputs (it does not matter for Watcher-Info that inputs are already spend)

Without additinal „sanity” spending on finalizations the outputs will persist.

Reply:
Regarding the explicit test for the scenario I think we have it splitted into separated (smaller) tests:
Pt 1 is covered,
Pt 2 - test showing that spending non-existing output does not break,
Pt 3 - test spending output by „finalization”-like event

I’ll add a comment in the test module to describe the scenario and it’s coverage by the tests already present